test/contracts/LLL_ERC20.cpp

```diff
@@ -0,0 +1,590 @@
+      ;; @param value The maximum allowed amount.
+      ;; @return Success (true). Other outcomes result in a Revert.
+
+      (function approve
+        (seq has-two-args (is-address arg1) (is-value arg2)
+
+          (def 'spender arg1)
+          (def 'value   arg2)
+
+          ;; Force users set the allowance to 0 before setting it to
+          ;; another value for the same spender. Prevents this attack:
+          ;; https://docs.google.com/document/d/1YLPtQxZu1UAvO9cZ1O2RPXBbT0mooh4DYKjA_jp-RLM
+          (when
+            (&& value (sload (allowance (caller) spender)))
+            (revert))
+          
+          (sstore (allowance (caller) spender) value)
+
+          ;; Event - Approval(address,address,uint256)
+          (event3 approval-event-id (caller) spender value)
+          
+          (return true)))
+      
+      ;; ----------------------------------------------------------------------
+      ;; Returns the amount which _spender is still allowed to withdraw
+      ;;   from _owner.
+      ;; @abi allowance(address,address) constant returns (uint256)
+      ;; @param owner The owning account.
+      ;; @param spender The withdrawing account.
+      ;; @return The allowed amount remaining.
+
+      (function get-allowance
+        (seq
+
+          (def 'owner   arg1)
+          (def 'spender arg2)
+          
+          (return (sload (allowance owner spender)))))
+      
+      ;; ----------------------------------------------------------------------
+      ;; Fallback: No functions matched the function ID provided.
+
+      (revert)))
+  )
+)DELIMITER";
+
+static unique_ptr<bytes> s_compiledErc20;
+
+class LllErc20TestFramework: public LLLExecutionFramework
+{
+protected:
+	void deployErc20()
+	{
+		if (!s_compiledErc20)
+		{
+			vector<string> errors;
+			s_compiledErc20.reset(new bytes(compileLLL(erc20Code, false, &errors)));
+		}
+		sendMessage(*s_compiledErc20, true);
+		BOOST_REQUIRE(!m_output.empty());
+	}
+
+	std::string abiString(bytes const& ABIString)
+	{
+		bytesConstRef ret = ref(ABIString);
+		BOOST_REQUIRE(ret.size() >= 0x20);
+		u256 offset = eth::abiOut<u256>(ret);
+		BOOST_REQUIRE_EQUAL(offset, 0x20);
+		u256 len = eth::abiOut<u256>(ret);
+		BOOST_REQUIRE_EQUAL(ret.size(), ((len + 0x1f) / 0x20) * 0x20);
+		return ret.cropped(0, size_t(len)).toString();
+	}
+};
+
+}
+
+// Test suite for an ERC20 contract written in LLL.
+BOOST_FIXTURE_TEST_SUITE(LLLERC20, LllErc20TestFramework)
+
+BOOST_AUTO_TEST_CASE(creation)
+{
+	deployErc20();
+
+	// All tokens are initially assigned to the contract creator.
+	BOOST_CHECK(callContractFunction("balanceOf(address)", ACCOUNT(0)) == encodeArgs(TOKENSUPPLY));
+}
+
+BOOST_AUTO_TEST_CASE(constants)
+{
+	deployErc20();
+
+	BOOST_CHECK(callContractFunction("totalSupply()") == encodeArgs(TOKENSUPPLY));
+	BOOST_CHECK(callContractFunction("decimals()") == encodeArgs(TOKENDECIMALS));
+	BOOST_CHECK(abiString(callContractFunction("symbol()")) == TOKENSYMBOL);
```

Comment:
Great! That's better.  I had to do, `encodeDyn(string(TOKENSYMBOL))` for reasons not clear, but it works.

Just committing an update.